src/clj/rems/application/eraser.clj

```diff
@@ -0,0 +1,53 @@
+(ns rems.application.eraser
+  (:require [mount.core :as mount]
+            [clj-time.core :as time]
+            [clojure.test :refer :all]
+            [clojure.tools.logging :as log]
+            [rems.scheduler :as scheduler]
+            [rems.db.applications :as applications]
+            [rems.config :refer [env]])
+  (:import [org.joda.time Period]))
+
+(defn- is-expired-draft?
+  [application]
+  (let [expiration-threshold (-> env :application-expiration :draft Period/parse)
+        last-activity (:application/last-activity application)
+        state (:application/state application)]
+    (and (= :application.state/draft state)
+         (time/before? last-activity (time/minus (time/now) expiration-threshold)))))
+
+(deftest test-is-expired-draft
+  (with-redefs [env {:application-expiration {:draft "P90D"}}]
+    (testing "should identify expired draft application"
+      (is (true? (is-expired-draft? {:application/state :application.state/draft
+                                     :application/last-activity (time/minus (time/now) (time/days 90) (time/seconds 1))})))
+      (is (false? (is-expired-draft? {:application/state :application.state/draft
+                                      :application/last-activity (time/now)})))
+      (is (false? (is-expired-draft? {:application/state :application.state/submitted
+                                      :application/last-activity (time/minus (time/now) (time/days 90))}))))))
+
+(defn remove-expired-applications! []
+  (log/info "Start rems.application.eraser/remove-expired-applications!")
```

Comment:
Should we add `Finished ...` logging too?

Btw. sometimes I use the `#'remove-expired-applications!` in the message so it "tracks" the fn with a bit more certainty. We haven't used that pattern in REMS I think.

Something like so:
```
(log/info :start #'remove-expired-applications!)
```

Reply:
nice! i was actually thinking about how to get the symbol of the executing fn in actual namespace. i tried all sorts of namespace functions but they didn't quite work out. :)

Reply:
There are ways around this. 
- This kind of start/end logging is sometimes called tracing and there are I think some libraries to instrument functions with the info more automatically (kind of like aspect oriented programming examples in Java). 
  - https://github.com/clojure/tools.trace
  - https://github.com/clojure-emacs/sayid
  - lots of ...
- One can also have custom loggers / defns that would do this automatically.
- There seems to exist also https://cloojure.github.io/doc/tupelo/tupelo.misc.html#var-fn-info.